lib/ood_core/job/adapters/linux_host/launcher.rb

```diff
@@ -57,7 +57,7 @@ def initialize(
   # @param hostname [#to_s] The hostname to submit the work to
   # @param script [OodCore::Job::Script] The script object defining the work
   def start_remote_session(script)
-    cmd = ssh_cmd(submit_host)
+    cmd = ssh_cmd((script.native && script.native['submit_host_override']) ? script.native['submit_host_override'] : submit_host)
```

Comment:
Instead of `attr_reader :submit_host` and then this line we should have a custom getter for submit host:

```ruby
def submit_host(script = nil)
  if script && script.native && script.native['submit_host_override']
    script.native['submit_host_override']
  else
    submit_host
  end
end
```

then we keep `cmd = ssh_cmd(submit_host)`.

Reply:
err we change that to `cmd = ssh_cmd(submit_host(script))`